### PR TITLE
Implement an admin HRM endpoint to validate a twilio flex JWE and return if user has permissions to perform action

### PR DESCRIPTION
**Pull request by GPaoloni**
## Description
This PR
- Adds logic to, given a user, a set of actions and an object, evaluate if the user can can perform the actions on the object. This is initially intended to use by the "get signed URL download" Lambda to retrieve transcripts/recordings, but is already intended to replace other in-place evaluation of permissions (like case endpoints).
- Exposes a GET endpoint `/permissions/:action`, which requires a valid Twilio JWT and accepts the following query parameters:
  - `objectType`: `TargetKind`
  - `objectId`: `number`
  - `bucket` (optional): `string`
  - `key` (optional): `string`
  which will evaluate the JWT. If it's valid, will perform a permission check with the JWT's user, the object (composing `objectType` and `objectId`) and the value given in `:action`, returning `200` status code if the action can be performed by the user or `403` if not.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-1782)
- [x] New tests added

### Verification steps

### CODE REVIEW

File: hrm-domain/hrm-service/permission-rules/br.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   "viewContact": [["everyone"]],
   "editContact": [["isSupervisor"]],
   "viewExternalTranscript": [["isSupervisor"]],
+  "viewRecording": [],
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/permission-rules/br.json:25]: @janorivera I'm not adding the actual permissions to the helplines. Can you take care on checking which permissions should be set for them? Or should I copy the `viewExternalTranscript` one?

File: hrm-domain/hrm-service/src/permissions/permissions-routes-v0.ts [hunk 2]
```diff
@@ -40,5 +50,90 @@ export default (permissions: Permissions) => {
...
+
+      if (isErrorResult(canPerformResult)) {
+        return next(createError(canPerformResult.statusCode, canPerformResult.message));
+      }
+
+      if (!canPerformResult.data) {
+        return next(createError(403, 'Not allowed'));
+      }
+
+      if (bucket && key) {
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/src/permissions/permissions-routes-v0.ts:109]: Tight to actual actions

**Review by robert-bo-davis — APPROVED** [review]
LGTM after cleanup we discussed in Pairing session

File: hrm-domain/hrm-service/permission-rules/br.json (hunk 1 above)
Comment (janorivera) [reply hrm-domain/hrm-service/permission-rules/br.json:25]: @GPaoloni OK I think CL and CA uses calls and probably ZW (?)
In any case I've asked Sinekhaya and based on what I know from CL I think is safe to assume that **viewRecording** can have the same value as **viewExternalTranscript**.  @dee-luo is not here to confirm about CA but I'm assuming that is only for supervisors.

**Pull request merged**